openassessment/xblock/validation.py

```diff
@@ -58,27 +62,37 @@ def _is_valid_assessment_sequence(assessments):
         bool
 
     """
-    valid_sequences = [
-        ['self-assessment'],
-        ['peer-assessment'],
-        ['peer-assessment', 'self-assessment'],
-        ['self-assessment', 'peer-assessment'],
-        ['student-training', 'peer-assessment'],
-        ['student-training', 'peer-assessment', 'self-assessment'],
-        ['student-training', 'self-assessment', 'peer-assessment'],
-        ['example-based-assessment'],
-        ['example-based-assessment', 'self-assessment'],
-        ['example-based-assessment', 'peer-assessment'],
-        ['example-based-assessment', 'peer-assessment', 'self-assessment'],
-        ['example-based-assessment', 'self-assessment', 'peer-assessment'],
-        ['example-based-assessment', 'student-training', 'peer-assessment'],
-        ['example-based-assessment', 'student-training', 'peer-assessment', 'self-assessment'],
-        ['example-based-assessment', 'student-training', 'self-assessment', 'peer-assessment'],
-    ]
```

Comment:
The previous code was correct, self then peer has been allowed in master since https://github.com/edx/edx-ora2/commit/0c0935ddb8d5a16d7ad41d4c05ee9b2b9cd65808.

There are a whole series of validation tests run on this logic, they basically try to load a series of valid/invalid json and verify that they are correctly accepted or rejected. See https://github.com/edx/edx-ora2/blob/efischer/enable_staff_grading/openassessment/xblock/test/test_validation.py#L21